pandas/core/groupby/generic.py

```diff
@@ -1569,6 +1574,198 @@ def func(df):
+                            "the groupby column keys"
+                        )
+                remaining_columns = subset
+
+            if dropna:
+                df = df.dropna(subset=remaining_columns, axis="index", how="any")
+
+            # Add the remaining_column keys to the main grouper
+            main_grouper, _, _ = get_grouper(
+                df,
+                key=self.keys,
+                axis=self.axis,
+                level=self.level,
+                sort=self.sort,
+                mutated=self.mutated,
+                dropna=self.dropna,
+            )
+            groupings = list(main_grouper.groupings)
+            for key in remaining_columns:
+                grouper, _, _ = get_grouper(
+                    df,
+                    key=key,
+                    axis=self.axis,
+                    level=self.level,
+                    sort=self.sort,
+                    mutated=self.mutated,
+                    # nulls have already been dropped from remaining_columns
+                    dropna=False,
+                )
+                groupings += list(grouper.groupings)
+
+            # Take the size of the overall columns
+            result = df.groupby(
+                groupings,
+                sort=self.sort,
+                observed=self.observed,
+                dropna=self.dropna,
+            ).size()
+            result.name = RESULT_NAME
+
+            if normalize:
+                # Normalize the results by dividing by the original group sizes
+                indexed_group_size = df.groupby(
+                    main_grouper,
+                    sort=self.sort,
+                    observed=self.observed,
+                    dropna=self.dropna,
+                ).size()
+                if non_column_grouping:
+                    # The common index needs a common name
+                    indexed_group_size.index.set_names("Group", inplace=True)
+                    result.index.set_names("Group", level=0, inplace=True)
+                # Use indexed group size series
+                if self.dropna or (isinstance(self.keys, list) and len(self.keys) == 1):
+                    result /= indexed_group_size
+                else:
+                    # Unfortunately, nans in multi-column multiindex sometimes makes
+                    # size() produce a Series that puts nans in the result
+                    values = (
+                        result.values
+                        / indexed_group_size.align(result, join="left")[0].values
+                    )
+                    result = Series(data=values, index=result.index, name=RESULT_NAME)
+                if non_column_grouping:
+                    result.index.set_names(None, level=0, inplace=True)
+
+            if sort:
+                # Sort the values and then resort by the main grouping
+                if non_column_grouping:
+                    level: Any = 0
```

Comment:
This assumes a single grouper as well, which may not be the case.

Reply:
Fixed